cmd/installer/pkg/install/install.go

```diff
@@ -112,68 +111,27 @@ func NewInstaller(cmdline *procfs.Cmdline, seq runtime.Sequence, opts *Options)
 }
 
 // Verify existence of boot partition.
-func (i *Installer) probeBootPartition() error {
-	// there's no reason to discover boot partition if the disk is about to be wiped
-	if !i.options.Zero {
-		dev, err := blockdevice.Open(i.options.Disk)
-		if err != nil {
-			i.bootPartitionFound = false
-
-			return err
-		}
-
-		defer dev.Close() //nolint:errcheck
-
-		if part, err := dev.GetPartition(constants.BootPartitionLabel); err != nil {
-			i.bootPartitionFound = false
-		} else {
-			i.bootPartitionFound = true
-
-			// mount the boot partition temporarily to find the bootloader labels
-			mountpoints := mount.NewMountPoints()
-
-			partPath, err := part.Path()
-			if err != nil {
-				return err
-			}
-
-			fsType, err := part.Filesystem()
-			if err != nil {
-				return err
-			}
-
-			mountpoint := mount.NewMountPoint(partPath, constants.BootMountPoint, fsType, unix.MS_NOATIME|unix.MS_RDONLY, "")
-			mountpoints.Set(constants.BootPartitionLabel, mountpoint)
-
-			if err := mount.Mount(mountpoints); err != nil {
-				log.Printf("warning: failed to mount boot partition %q: %s", partPath, err)
-			} else {
-				defer mount.Unmount(mountpoints) //nolint:errcheck
-			}
-		}
+func (i *Installer) probeBootloader() error {
+	config, err := grub.Probe()
+	if err != nil {
+		return err
 	}
 
-	grubConf, err := grub.Read(grub.ConfigPath)
+	i.Current, i.Next, err = config.Flip()
 	if err != nil {
 		return err
 	}
 
-	next := grub.BootA
+	i.bootLoaderFound = true
```

Comment:
```suggestion
	i.bootLoaderFound = config != nil
```

Maybe? And you can remove `i.bootLoaderFound = false` below.